hepcrawl/spiders/mit_spider.py

```diff
@@ -0,0 +1,243 @@
+            except AttributeError:
+                pass
+            surname, given_names = split_fullname(author)
+            authdict = {
+                'surname': surname,
+                'given_names': given_names,
+            }
+            if affiliation:
+                authdict["affiliations"] = [{"value": affiliation}]
+            authors.append(authdict)
+
+        return authors
+
+    def add_fft_file(self, pdf_files, file_access, file_type):
+        """Create a structured dictionary to add to 'files' item."""
+        file_dicts = []
+        for link in pdf_files:
+            file_dict = {
+                "access": file_access,
+                "description": self.name,
+                "url": urljoin(self.domain, link),
+                "type": file_type,
+            }
+            file_dicts.append(file_dict)
+        return file_dicts
+
+    @staticmethod
+    def get_thesis_info(node):
+        """Create thesis info dictionary."""
+        date = node.xpath(
+            "//td[contains(text(), 'dc.date.issued')]/following-sibling::td[1]/text()").extract_first()
+        university = node.xpath(
+            "//td[contains(text(), 'dc.publisher')]/following-sibling::td[1]/text()").extract_first()
+
+        thesis = {
+            "date": date,
+            "university": university,
+            "degree_type": "PhD",
+        }
+        return thesis
+
+    @staticmethod
+    def get_thesis_supervisors(node):
+        """Create a structured supervisor dictionary.
+
+        There might be multiple supervisors.
+        """
+        supervisors_raw = node.xpath(
+            "//td[contains(text(), 'dc.contributor.advisor')]/following-sibling::td[1]/text()").extract()
+        supers = []
+        for supervisor in supervisors_raw:
+            if "and" in supervisor:
+                supers.extend(supervisor.split(" and "))
+            else:
+                supers.append(supervisor)
+
+        supervisors = []
+        for supervisor in supers:
+            supervisors.append({
+                'raw_name': supervisor,
+                'affiliation': "",
+            })
+
+        return supervisors
+
+    @staticmethod
+    def get_page_nr(node):
+        """Get and format the page numbers. Return only digits."""
+        page_nr_raw = node.xpath(
+            "//td[contains(text(), 'dc.format.extent')]/following-sibling::td[1]/text()").extract_first()
+        if page_nr_raw:
+            return ''.join(i for i in page_nr_raw if i.isdigit())
+
+    def parse_node(self, response, node):
+        """Parse MIT thesis listing and find links to record splash pages."""
+        link = node.xpath(
+            ".//div[@class='artifact-title']/a/@href").extract_first()
+        splash_link = urljoin(self.domain, link) + "?show=full"
+        if splash_link:
+            request = Request(splash_link, callback=self.scrape_splash)
+            yield request
+
+    def scrape_splash(self, response):
```

Comment:
Shall we maybe combine the functions `scrape_splash` and `build_item` into one `build_item` as `scrape_splash` do not yield any new requests anyway. Then you could even use `add_xpath` for some values.